File: 2026-09/src/day_13/leet_268/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
   public:
    int missingNumber(vector<int>& nums) {
        int n = nums.size();
        vector<bool> seen(1 + n);
        for (int x : nums) seen[x] = true;

        for (int i = 0; i <= n; i++) {
            if (!seen[i]) return i;
        }

        return -1;
    }
};

File: 2026-09/src/day_13/leet_268/Solution.java
package day_13.leet_268;

class Solution {
    public int missingNumber(int[] nums) {
        int n = nums.length;
        boolean[] seen = new boolean[1 + n];
        for (int x : nums) {
            seen[x] = true;
        }

        for (int i = 0; i <= n; i++) {
            if (!seen[i]) return i;
        }

        return -1;
    }
}
